### PR TITLE
Freeradius: add parser

### PR DESCRIPTION
**Pull request by squioc**
Add a parser and smart-descriptions for FreeRADIUS authentication logs.

### CODE REVIEW

File: FreeRADIUS/freeradius/ingest/parser.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          event.category: '{{ parsing.message.event_category }}'
+          event.reason: '{{ parsing.message.event_reason }}'
+          user.email: '{{ parsing.message.user_email }}'
+          user.name: '{{ parsing.message.user_name }}'
+          user.domain: '{{ parsing.message.user_domain }}'
+          network.name: '{{ parsing.message.network_name }}'
+          client.port: '{{ parsing.message.client_port }}'
+          client.mac: '{{ parsing.message.client_mac }}'
+          client.ip: '{{ parsing.message.client_ip }}'
+          client.domain: '{{ parsing.message.client_domain }}'
```
Comment (PierrickV) [inline FreeRADIUS/freeradius/ingest/parser.yml:36]:
```suggestion
          source.port: '{{ parsing.message.client_port }}'
          source.mac: '{{ parsing.message.client_mac }}'
          source.ip: '{{ parsing.message.client_ip }}'
          source.domain: '{{ parsing.message.client_domain }}'
```

I suggest that we use `source` as we do for other parsers

File: FreeRADIUS/freeradius/ingest/parser.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  - name: set_ecs_fields
+stages:
+  set_ecs_fields:
+    actions:
+      - set:
+          event.kind: 'event'
+          event.category: ['authentication']
+          event.type: ['info']
+          event.dataset: 'freeradius.authentication'
+          event.category: '{{ parsing.message.event_category }}'
```
Comment (PierrickV) [inline FreeRADIUS/freeradius/ingest/parser.yml:27]: it would be interesting to have an event with this field in test files

**Comment by PierrickV** [comment]
Thank you for this work!

Would it be possible for you to update the [manifest](https://github.com/SEKOIA-IO/intake-formats/blob/main/FreeRADIUS/_meta/manifest.yml) ? ATM there is no description and no data_sources

File: FreeRADIUS/freeradius/ingest/parser.yml (hunk 1 above)
Comment (squioc) [reply FreeRADIUS/freeradius/ingest/parser.yml]: Done

File: FreeRADIUS/freeradius/ingest/parser.yml (hunk 2 above)
Comment (squioc) [reply FreeRADIUS/freeradius/ingest/parser.yml]: I removed this fields that override the previous one (L24)

File: FreeRADIUS/freeradius/_meta/smart-descriptions.json [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+          "field": "network.name"
+      },
+      {
+        "field": "freeradius.outcome",
+        "value": "Login OK"
+      }
+    ]
+  },
+  {
+    "value": "The user {user.name} logged in from {network.name} with mac {source.mac} and port {source.port}",
```
Comment (PierrickV) [inline FreeRADIUS/freeradius/_meta/smart-descriptions.json:24]: Just out of curiosity, how is it possible to have a `source.port` and a missing `source.ip` ? :thinking:

File: FreeRADIUS/freeradius/_meta/smart-descriptions.json [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    "conditions": [
+      {
+        "field": "user.name"
+      },
+      {
+        "field": "source.mac"
+      },
+      {
+          "field": "source.port"
```
Comment (PierrickV) [inline FreeRADIUS/freeradius/_meta/smart-descriptions.json:33]: You could format the file globally

File: FreeRADIUS/freeradius/ingest/parser.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+name: untitled-name
```
Comment (PierrickV) [inline FreeRADIUS/freeradius/ingest/parser.yml:1]: Empty name here (where is this field used BTW?)

**Review by PierrickV — APPROVED** [review]
LGTM !

File: FreeRADIUS/freeradius/ingest/parser.yml (hunk 5 above)
Comment (squioc) [reply FreeRADIUS/freeradius/ingest/parser.yml]: Fixed

File: FreeRADIUS/freeradius/_meta/smart-descriptions.json (hunk 3 above)
Comment (squioc) [reply FreeRADIUS/freeradius/_meta/smart-descriptions.json:24]: Yes. For application authentication, the logs only supply mac address and port (see [this example](https://github.com/SEKOIA-IO/intake-formats/blob/feat/FreeRadius/FreeRADIUS/freeradius/tests/test_login_ok1.json#L9))

**Review by PierrickV — APPROVED** [review]

**Pull request merged**